Sources/SmithyHTTPAuth/SigV4AuthScheme.swift

```diff
@@ -34,8 +34,9 @@ public struct SigV4AuthScheme: AuthScheme {
             value: context.isBidirectionalStreamingEnabled
         )
 
-        // Set signing name and signing region flags
-        updatedSigningProperties.set(key: SigningPropertyKeys.signingName, value: context.signingName)
+        // Set resolved signing name and signing region flags
+        let signingName = updatedSigningProperties.get(key: SigningPropertyKeys.signingName) ?? context.signingName
```

Comment:
If signing properties include a signing name, use it, else fall back to the default signing name defined in the model's sigv4 trait.